shm/Map.cpp

```diff
@@ -1,7 +1,7 @@
 #include "Map.hpp"
 #include <algorithm>
 
-coordinateType generatePosition() {
+Island::coordinateType generatePosition() {
```

Comment:
line 29 in Map.ccp - move islandVec_.reserve(...) from c-tor to fillWithRandomIslands() - it was Mateusz suggestion

Reply:
I think this PR is to fix Island and Coordinates clases only. Here the change is only to access coordinateType which is now in the Island class as Mateusz suggested. For me this conversation can be resolved as I consider that everything is OK here.